app/views/smart_proxies/plugins/_puppet_envs.html.erb

```diff
@@ -0,0 +1,20 @@
+<% if envs.present? %>
```

Comment:
Render an `empty_state`, for instance the partial `environments/help` when envs is not present too. I think it'd be clearer to do this at the controller level though.